objects/69ead2/contained/DUESETUP/script.lua

```diff
@@ -0,0 +1,643 @@
+        max_distance = 1,
+        -- debug = true
+    })
+    for _,hit in pairs(hits) do
+        if hit.hit_object.type == "Card" and hit.hit_object.getVar("energy") ~= nil then -- check cards for energy to make sure they are correct hits
+            local card = hit.hit_object
+            if impendTable[card.guid] == nil then
+                local costs = {}
+                costs[card.guid] = card.getVar("energy")
+                local energy = Global.call("modifyCost",{color = color, costs = costs})[card.guid]
+                card.hide_when_face_down = false
+                impendTable[card.guid] = {
+                    guid = card.guid,
+                    name = card.getName(),
+                    maxEnergy = energy, -- may need to revert to storing without discount
+                    turnSelected = Global.getVar("turn"),
+                    energy = {0,0,0},
+                }
+            end
+            if impendTable[card.guid] then -- don't think this is necessary any more, but useful is a case arises
+                createCardButtons(card)
+                updateEnergyDisplay(card.guid)
+                card.deal(1, color, 3)
+            end
+            -- print(card.guid .. " is impended") -- debug
+        end
+    end
+    updateSave()
+end
+
+function g3Toggle(obj)
+    local ind = getButtonIndex(obj,"g3Toggle")
+    if g3Selected then
+        g3Selected = false
+        obj.editButton({
+            index = ind,
+            label = "☐\n\n\n",
+        })
+        removeAllG3Buttons()
+    else    
+        g3Selected = true
+        obj.editButton({
+            index = ind,
+            label = "☑\n\n\n",
+        })
+        createAllG3Buttons()
+    end
+    updateSave()
+end
+
+--------------------------------------
+-- impended card buttons and functions
+--------------------------------------
+
+function createCardButtons(card)
+    if not getButtonIndex(card,"energyDisplay") then -- stop multiple buttons appearing, did prevent another way before but that removed other functionality
+        card.createButton({ 
+            click_function = "energyDisplay",
+            function_owner = self,
+            label = "0/0",
+            position = {0.29,aboveCard,-0.55},
+            width = 700,
+            height = 550,
+            font_size = 430,
+            color = {0,0,0,buttonVisibility},
+            font_color = {1,1,1,100},
+            tooltip = "Select a card to impend",
+        })
+        
+        card.createButton({
+            click_function = "overridePlus",
+            function_owner = self,
+            label = "+",
+            position = {-0.2,-aboveCard,0.8},
+            rotation = {180,180,0},
+            width = 200,
+            height = 200,
+            font_size = 250,
+            color = {0,0,0,0.9},
+            font_color = {1,0,0,100},
+            tooltip = "Press this to override the value of impended energy.\nUseful for if time passes twice.",
```

Comment:
the `+` and `-` buttons are also meant to be used when you're paying non zero amount of energy when you impend the card right? or is there some code somewhere else to handle starting at non zero number?

Reply:
I won't be answering most of the questions about Dances, 'cause Lolzax wrote this code, but I can answer this one.

Right or left clicking (to increment or decrement respectively) the energy number on the front of the card is used to start with a non-zero amount of energy when you impend the card. But this can only be done on the turn where you're impending the card, hence the existence of the override buttons if you really need to change it on a later turn.

Reply:
Those buttons (on the back) are purely there as an override mechanism for cases like if the player accidentally double presses "time passes" (which I have myself a handful of times) and then the player needs to adjust the energy on the card.
The way to adjust the initial energy on the card upon impending it is to left/right click the face of the counter itself (left click to increment, right click to decrement). If it's preferable to make more buttons to make the system more intuitive to the user, that's fine, but I'd probably need some input on the design/location of such buttons. I tried to make these as non-intrusive as possible, not covering any needed information on any cards, meaning there isn't much room for placement of buttons.

Reply:
isn't there some protection in time passes where you can't hit the button again until the previous time passes operation is done (aka the quote has shown up)? where's the code for the buttons on the +/- button on the front of the card?

Reply:
The code for paying/unpaying energy when you impend the card (by clicking the button that shows the current energy) is in `energyDisplay()`. The code for the +/- buttons that appear on the front when you use G3 are in `g3Plus()` and `g3Minus()`.